components/scripts/gradle/gradle-init-scripts/configure-remote-build-caching.gradle

```diff
@@ -4,23 +4,110 @@ static getInputParam(Gradle gradle, String name) {
     return gradle.startParameter.systemPropertiesArgs[name] ?: System.getProperty(name) ?: System.getenv(envVarName)
 }
 
+def isTopLevelBuild = !gradle.parent
+
+def expDir = getInputParam(gradle, 'develocity.build-validation.expDir')
+def remoteBuildCacheType = getInputParam(gradle, 'develocity.build-validation.remoteBuildCacheType')
 def remoteBuildCacheUrl = getInputParam(gradle, 'develocity.build-validation.remoteBuildCacheUrl')
 
-settingsEvaluated { settings ->
+def docsRoot = 'https://docs.gradle.com/develocity/gradle-plugin'
+
+settingsEvaluated { Settings settings ->
     settings.buildCache {
         local {
             enabled = false
         }
-        remote(HttpBuildCache) {
-            enabled = true
-            push = false
-            if (remoteBuildCacheUrl) {
-                url = withPathTrailingSlash(new URI(remoteBuildCacheUrl))
+
+        if (remoteBuildCacheType) {
+            if (!['develocity', 'gradle-enterprise', 'http'].contains(remoteBuildCacheType)) {
+                // The scripts already fail if the value of --remote-build-cache-type isn't valid.
+                // This is for the sake of completeness since this init script assumes it's valid.
+                def errorFile = new File(expDir, 'errors.txt')
+                def message = "Invalid value '${remoteBuildCacheType}' for remote build cache type. Values are 'develocity', 'gradle-enterprise', or 'http'."
+                errorFile.text = message
+                throw new IllegalStateException(message)
+            }
```

Comment:
>  There is now a lot of code in this block, and a lot of nested if statements. I think it would be helpful to extract some of the nested if statements into helper functions so that the code is overall easier to read.

For example, this code block could be extracted into a `validateRemoteBuildCacheType` function.